src/dmd/hdrgen.d

```diff
@@ -963,6 +969,9 @@ public:
             buf.writeByte(')');
         }
         parametersToBuffer(t.parameters, t.varargs);
+        buf.writeByte(' ');
```

Comment:
This should be inside the if statement, otherwise you will have a space even if you don't have a `return` attribut. That's what makes the tests to fail.